### PR TITLE
OSOE-729: Prevent Azure-related workflows running at the same time when interacting with the same environment(s)

### LINKED ISSUE #281: Prevent Azure-related workflows running at the same time when interacting with the same environment(s) (OSOE-729)
For example, running `deploy-to-azure-app-service` and `reset-azure-environment` against the same environment will cause them to step on each other's toes (and fail or do something unexpected), same goes for any of these two and `swap-azure-web-app-slots` (also fix the O1 variants).

Update the [concurrency](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#concurrency) group names to be dynamic by using the slot name parameters.

[Jira issue](https://lombiq.atlassian.net/browse/OSOE-729)

### PR DESCRIPTION
**Pull request by BenedekFarkas**
[OSOE-729](https://lombiq.atlassian.net/browse/OSOE-729)
Fixes #281

[OSOE-729]: https://lombiq.atlassian.net/browse/OSOE-729?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Docs/Workflows/Index.md
Comment (BenedekFarkas) [inline Docs/Workflows/Index.md:1]: The only content change in this file is the categories section.

File: Docs/Workflows/BuildDotNetCoreOrchardCore.md
Comment (BenedekFarkas) [inline Docs/Workflows/BuildDotNetCoreOrchardCore.md:1]: Just factored out from the earlier Workflows.md and changed the second-level titles.

File: Docs/Workflows/AzureHosting.md
Comment (BenedekFarkas) [inline Docs/Workflows/AzureHosting.md:1]: Just factored out from the earlier Workflows.md and changed the second-level titles. Also moved common notes to the primary section.

File: Docs/Actions.md
Comment (BenedekFarkas) [inline Docs/Actions.md:1]: Text didn't change, just added categories.

File: Docs/Workflows/Productivity.md
Comment (BenedekFarkas) [inline Docs/Workflows/Productivity.md:1]: Just factored out from the earlier Workflows.md and changed the second-level titles.

File: Docs/Workflows/Index.md
Comment (Piedone) [reply Docs/Workflows/Index.md:1]: This breaks incoming links (we also have those in our wiki). This is necessary, but at that point, let's make it future proof: put all the workflows' docs into their own pages, and then you can elevate the content of this page (with the general notes and links to workflows' pages) to the Readme, so people have to dig less.

Comment (Psichorex) [reply Docs/Workflows/Index.md:1]: Looks good to me after the refactoring but I @Piedone please approve if you are fine as its your request.

Comment (Piedone) [reply Docs/Workflows/Index.md:1]: I'm not fine, hence my comment :).

Comment (Psichorex) [reply Docs/Workflows/Index.md:1]: Wait didn't this got addressed by the factored out .mds?
Edit: I see now, my brain got stuck at the new .mds.

Comment (BenedekFarkas) [reply Docs/Workflows/Index.md:1]: Re-added the original Workflow.md with the complete Workflows "table of contents", I think that's a good middle-ground between information accessibility and density, I'd rather not put all this into the Readme. Folder structure is according to the categorization in Workflow.md.

Comment (Piedone) [reply Docs/Workflows/Index.md:1]: Look good. The link in the Readme is broken BTW.

Comment (BenedekFarkas) [reply Docs/Workflows/Index.md:1]: Yeah, I noticed, went around just now and checked/fixed every link.

Comment (Psichorex) [reply Docs/Workflows/Index.md:1]: All the links seems to be fine, at least on my side so probably overall aswell. Went throught manually on most of them and I yet to find any broken ones.

**Review by Psichorex — APPROVED** [review]

**Comment by Psichorex** [comment]
I will merge if @Piedone approves. Everything seems fine. Approved by me.

**Comment by Piedone** [comment]
It's only up to you.

**Comment by Psichorex** [comment]
> It's only up to you.

I asked because you had an unresolved comment containing a request for which I didn't see a clear indication that it was resolved.

File: Docs/Workflows/Index.md
Comment (Psichorex) [reply Docs/Workflows/Index.md:1]: > Yeah, I noticed, went around just now and checked/fixed every link.

From this reply I thought that all links are fine now but for me SpellChecking.md has broken links.
https://github.com/Lombiq/GitHub-Actions/blob/763adec7ca1b54e16985554d97b88b253196b96f/Docs/Workflows/Productivity/SpellChecking.md#:~:text=proxied%20by%20the-,spelling%20action,-in%20this%20repository

**Comment by Piedone** [comment]
I understand, but it's still up to you, me doing a second review of this PR is unnecessary.

File: Docs/Workflows/Index.md
Comment (BenedekFarkas) [reply Docs/Workflows/Index.md:1]: Fixed!

**Review by Psichorex — APPROVED** [review]

**Pull request merged**
